cmd/issuer_initializer/main.go

```diff
@@ -43,9 +38,28 @@ func main() {
 		return
 	}
 
-	vaultCli, err := providers.NewVaultClient(cfg.KeyStore.Address, cfg.KeyStore.Token)
+	var vaultCli *vault.Client
+	if cfg.VaultUserPassAuthEnabled {
+		vaultCli, err = providers.NewVaultClientWithUserPassAuth(ctx, cfg.KeyStore.Address, cfg.VaultUserPassAuthPassword)
+		if err != nil {
+			log.Error(ctx, "cannot init vault client with Kubernetes Auth: ", "err", err)
+			return
+		}
+	} else {
+		vaultCli, err = providers.NewVaultClient(cfg.KeyStore.Address, cfg.KeyStore.Token)
+		if err != nil {
+			log.Error(ctx, "cannot init vault client: ", "err", err)
+			return
+		}
+	}
```

Comment:
This code seems to be repeated in some places in the PR. I think we could create an internal package with a constructor that accepts the config and returns a vault client. Something similar to redis.Open() or kms.Open()